receiver/nsxtreceiver/internal/metadata/generated_metrics_v2.go

```diff
@@ -0,0 +1,710 @@
+	metricsBuffer                  pmetric.Metrics   // accumulates metrics data before emitting.
+	metricNsxtInterfacePacketCount metricNsxtInterfacePacketCount
+	metricNsxtInterfaceThroughput  metricNsxtInterfaceThroughput
+	metricNsxtNodeCacheMemoryUsage metricNsxtNodeCacheMemoryUsage
+	metricNsxtNodeCPUUtilization   metricNsxtNodeCPUUtilization
+	metricNsxtNodeDiskUsage        metricNsxtNodeDiskUsage
+	metricNsxtNodeDiskUtilization  metricNsxtNodeDiskUtilization
+	metricNsxtNodeMemoryUsage      metricNsxtNodeMemoryUsage
+}
+
+// metricBuilderOption applies changes to default metrics builder.
+type metricBuilderOption func(*MetricsBuilder)
+
+// WithStartTime sets startTime on the metrics builder.
+func WithStartTime(startTime pcommon.Timestamp) metricBuilderOption {
+	return func(mb *MetricsBuilder) {
+		mb.startTime = startTime
+	}
+}
+
+func NewMetricsBuilder(settings MetricsSettings, options ...metricBuilderOption) *MetricsBuilder {
+	mb := &MetricsBuilder{
+		startTime:                      pcommon.NewTimestampFromTime(time.Now()),
+		metricsBuffer:                  pmetric.NewMetrics(),
+		metricNsxtInterfacePacketCount: newMetricNsxtInterfacePacketCount(settings.NsxtInterfacePacketCount),
+		metricNsxtInterfaceThroughput:  newMetricNsxtInterfaceThroughput(settings.NsxtInterfaceThroughput),
+		metricNsxtNodeCacheMemoryUsage: newMetricNsxtNodeCacheMemoryUsage(settings.NsxtNodeCacheMemoryUsage),
+		metricNsxtNodeCPUUtilization:   newMetricNsxtNodeCPUUtilization(settings.NsxtNodeCPUUtilization),
+		metricNsxtNodeDiskUsage:        newMetricNsxtNodeDiskUsage(settings.NsxtNodeDiskUsage),
+		metricNsxtNodeDiskUtilization:  newMetricNsxtNodeDiskUtilization(settings.NsxtNodeDiskUtilization),
+		metricNsxtNodeMemoryUsage:      newMetricNsxtNodeMemoryUsage(settings.NsxtNodeMemoryUsage),
+	}
+	for _, op := range options {
+		op(mb)
+	}
+	return mb
+}
+
+// updateCapacity updates max length of metrics and resource attributes that will be used for the slice capacity.
+func (mb *MetricsBuilder) updateCapacity(rm pmetric.ResourceMetrics) {
+	if mb.metricsCapacity < rm.ScopeMetrics().At(0).Metrics().Len() {
+		mb.metricsCapacity = rm.ScopeMetrics().At(0).Metrics().Len()
+	}
+	if mb.resourceCapacity < rm.Resource().Attributes().Len() {
+		mb.resourceCapacity = rm.Resource().Attributes().Len()
+	}
+}
+
+// ResourceOption applies changes to provided resource.
+type ResourceOption func(pcommon.Resource)
+
+// WithNsxtInterfaceID sets provided value as "nsxt.interface.id" attribute for current resource.
+func WithNsxtInterfaceID(val string) ResourceOption {
+	return func(r pcommon.Resource) {
+		r.Attributes().UpsertString("nsxt.interface.id", val)
+	}
+}
+
+// WithNsxtNodeID sets provided value as "nsxt.node.id" attribute for current resource.
+func WithNsxtNodeID(val string) ResourceOption {
+	return func(r pcommon.Resource) {
+		r.Attributes().UpsertString("nsxt.node.id", val)
```

Comment:
would it be appropriate to use [host](https://github.com/open-telemetry/opentelemetry-specification/blob/7e35bec9401d9bc970a7107af45906c0890040b6/specification/metrics/semantic_conventions/system-metrics.md#systemfilesystem---filesystem-metrics) resource attributes to describe an nsx-t node?

Reply:
```suggestion
		r.Attributes().UpsertString("host.id", val)
```

Just to confirm based off [this doc](https://github.com/open-telemetry/opentelemetry-specification/blob/33e51e72276c4aa1dbe4fba84b4b05944ac5c241/specification/resource/semantic_conventions/host.md) 

are you suggesting something like this?

Reply:
Yes.  But only if you think that makes sense (i.e. would most users consider an nsxt node a "host"?)

Reply:
Hmm I don't necessarily think I've heard them referred to as hosts, more often the conventional term has been node. The term "host" generally makes me think of ESXi systems rather than NSX nodes.

Reply:
Reading the docs, I agree.  Lets leave it as-is